app/src/main/java/com/github/capntrips/kernelflasher/ui/screens/slot/SlotViewModel.kt

```diff
@@ -125,7 +125,8 @@ class SlotViewModel(
                     else -> log(context, "Invalid boot.img", shouldThrow = true)
                 }
             } else {
-                log(context, "Invalid boot.img", shouldThrow = true)
+                // boot.img v4 has no ramdisk!
+                _sha1 = Shell.cmd("$magiskboot sha1 $boot").exec().out.firstOrNull()
```

Comment:
It checks for `init_boot` [here](https://github.com/capntrips/KernelFlasher/blob/v1.0.0-alpha14/app/src/main/java/com/github/capntrips/kernelflasher/ui/screens/main/MainViewModel.kt#L58). In the case where `magiskboot` can't unpack the ramdisk, it probably makes sense to handle it in some way, rather than silently ignoring it. Maybe by making the getter [here](https://github.com/capntrips/KernelFlasher/blob/v1.0.0-alpha14/app/src/main/java/com/github/capntrips/kernelflasher/ui/screens/slot/SlotViewModel.kt#L73-L74) nullable, displaying an error [here](https://github.com/capntrips/KernelFlasher/blob/v1.0.0-alpha14/app/src/main/java/com/github/capntrips/kernelflasher/ui/components/SlotCard.kt#L38-L46), and adding some form of warning dialog. I can take a look at it in the next few days if it isn't something you want to tackle immediately. I already have warning dialogs on my radar from [here](https://forum.xda-developers.com/t/dev-kernel-flasher.4439887/post-88254575).